### DISCUSSION TITLE
how to use ths file of natvis in vscode and xmake？

### DISCUSSION THREAD

**Discussion by naonao-cola** [nlohmann/json#4367] [Q&A]
`
  "xmake.customDebugConfig": {
    "console": "integratedTerminal", // XMake调试时使用集成终端而非 debug console,也可以使用 externalTerminal
    "visualizerFile": "${workspaceFolder}/nlohmann_json.natvis",
    //"showDisplayString": true,
  },
`
我在vscode里面这样设置，使用xmake 编译项目，使用codelldb 进行调试。看到的json 不是键值对，而是my_type 之类的很深的层次。该如何解决

**Comment by nlohmann** [comment]
This looks like a debugger/IDE integration limitation rather than a library issue: .natvis files (this project generates nlohmann_json.natvis via tools/generate_natvis) are Microsoft's Visual-Studio-native-debugger visualization format. They work as expected with the MSVC debug engine (cppvsdbg in VS Code), but codelldb wraps LLDB rather than Microsoft's debug engine and has only partial/experimental Natvis support, so it commonly falls back to showing raw internal fields (like basic_json's inline ABI-namespace layout) instead of a friendly key/value view. This can't be fixed from the natvis file alone; switching to the cppvsdbg debug engine where available, or checking codelldb's own Natvis support/version, are the next things to try.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
